proposals/LP0004-project-governance.md

```diff
@@ -0,0 +1,393 @@
+
+When acting to facilitate decision making the _area team _should act as a mediator between different perspectives helping find common ground and recognizing that decisions need not be binary. The _area team_ should seek to find the best solution to the framed problem, which may not be any of the proposed alternatives. If agreement cannot be reached, the _area team_ may act as the final decision maker. In that capacity decisions of an _area team_ are considered final, but can be overruled by a 2/3 majority vote of the _project council_ or the _area team_ itself revisiting the issue.
+
+_Area teams_ are not intended to be direction setters or primary maintainers of their areas, although individuals on an _area team_ may fulfill that role separately. The _area team’s_ role is as a steward and moderator ensuring the health and smooth operation of the area.
+
+Each _area team_ will have 5 members elected by the _voting contributors_. Candidates for _area teams_ must be a _voting contributor_ and self-nominated. An individual cannot serve on two _area teams_.
+
+Members of an _area team_ are elected for 1 year terms. Individuals are encouraged to not run for more than two consecutive terms. To encourage turnover, incumbent members of an _area team_ must place in the top 4 candidates during an election. The fifth spot will always go to a non-incumbent candidate.
+
+
+#### Initial Area Teams
+
+With this proposal a group of initial _area teams_ will be formed with defined domain areas. The members of each _area team_ will be defined by an election held after the adoption of the proposal. The initial _area teams_ proposed below are based on statistics generated from the LLVM project repository (See Appendix 2 for applicable scripts).
+
+In dividing the LLVM project into _area teams_, this proposal seeks to break up areas of the project based on contribution volume and number of active contributors. This proposal assumes that areas of the project with more unique contributors are more likely to have contributor disagreements, and that areas with more highly active contributors will be more likely to have diverse _area team_ formations.
+
+This proposal puts a possible set of _area teams_ to form with the specified areas of ownership. One notable point is that LLVM (`:/llvm`) had 945 unique authors contribute over the 12 month period used for statistics. This is far more than any other part of the project, and it justified splitting LLVM into two teams, which are separated as the _LLVM Backend_ and _LLVM Middle-end_. Clang had 706 unique authors, which is also significantly higher than other sub-projects. To represent different parts of the Clang community the _Clang_ and _Clang Tooling_ teams are separated below.
+
+* LLVM Backend (CodeGen, MC, Target)
+* LLVM Middle-end (Everything not covered by LLVM Backend)
```

Comment:
LLVM has a lot of stuff that cannot be categorized into backend or middle-end: ADT/Support, IR, testing tools
My personal feeling is that LLVM components are more diversed and why selecting a code owner is extremely challenging. `.github/new-prs-labeler.yml` may demonstrate how diverse `llvm/` is.

I understand that it can be unwieldy to have too many LLVM area teams. A thought is to add a `LLVM Others` or something like that.

Reply:
I second in particular the point that it should be made very clear which area the evolution of IR itself falls into, and I second the notion that it probably make sense to put it together with some of the other examples inside a dedicated area, though "Others" sounds a bit strange to my ears. Maybe "LLVM Base" or "LLVM Core" or something like that?

Reply:
There are definitely flaws in the metrics I used to divide the teams. Based on the total number of unique contributors to LLVM (945), I felt comfortable splitting it in two, but not three. I was trying to keep clang (706) and clang-tools-extra (159) as two teams by slicing a bit of clang into clang-tools-extra. There is a _huge_ flaw in my scripts that the unique contributors to clang and clang-tools-extra isn't 865 because my script doesn't account for the overlap in contributors (which I suspect is significant). That flaw also applies to LLVM's sub-directories.

I'm going to play around with making my scripts a bit smarter to see if I can get some better ideas how to split things by activity. For LLVM & Clang, where the projects are getting split, I think I'll need to be able to come up with an explicit filtering for which directories to include and exclude (if only our community had grown in the shape of our source repo 😜).

Reply:
I would not try to separate per number of people, but per skillset. I don't see a problem in an area team for 40 people in one side and 400 in the other...

Reply:
I'm very open for alternative suggestions on the breakdown of area teams.